backend-shared/config/db/ephemeralPostgres.go

```diff
@@ -0,0 +1,197 @@
+		psqlcmd.Env = os.Environ()
+		psqlcmd.Env = append(psqlcmd.Env, "PGPASSWORD=gitops")
+		var outb, errb bytes.Buffer
+		psqlcmd.Stdout = &outb
+		psqlcmd.Stderr = &errb
+
+		psqlErr := psqlcmd.Run()
+
+		if errb.String() != "" {
+			return false, fmt.Errorf(errb.String())
+		}
+		if psqlErr != nil {
+			return false, psqlErr
+		}
+		fmt.Printf("%s database is ready to use\n", tempDBName)
+		return true, nil
+	})
+	if errWait != nil {
+		return EphemeralDB{}, fmt.Errorf("error in connecting to the database: %s", errWait)
+	}
+
+	newDBName := "postgres"
+	dbcmd = "PGPASSWORD=gitops psql -h localhost -d %s -U postgres -p 6432"
+	fmt.Println("\nRunning: ", fmt.Sprintf(dbcmd, newDBName))
+
+	psqlcmd := exec.Command("psql", "-h", "localhost", "-d", newDBName, "-U", "postgres", "-p", "6432")
+	psqlcmd.Env = os.Environ()
+	psqlcmd.Env = append(psqlcmd.Env, "PGPASSWORD=gitops")
+	var errConnection bytes.Buffer
+	psqlcmd.Stderr = &errConnection
+	psqlErr := psqlcmd.Run()
+	if errConnection.String() != "" {
+		return EphemeralDB{}, fmt.Errorf("Error in creating a new database: %s", errConnection.String())
+	}
+	if psqlErr != nil {
+		return EphemeralDB{}, fmt.Errorf("Error in creating a new database: %s", psqlErr)
+	}
+
+	fmt.Printf("the %s database is created and ready to use\n", newDBName)
+
+	// Following command is used to populate the database tables from the db-schema.sql (defined in the monorepo)
+	dbcmd = "PGPASSWORD=gitops psql -h localhost -d %s -U postgres -p 6432 -q -f ../../../db-schema.sql"
+	fmt.Println("\nRunning: ", fmt.Sprintf(dbcmd, newDBName))
+	psqlcmd = exec.Command("psql", "-h", "localhost", "-d", newDBName, "-U", "postgres", "-p", "6432", "-q", "-f", "../../../db-schema.sql")
+	var errSchema bytes.Buffer
+	psqlcmd.Stderr = &errSchema
+	psqlcmd.Env = os.Environ()
+	psqlcmd.Env = append(psqlcmd.Env, "PGPASSWORD=gitops")
+	psqlErr = psqlcmd.Run()
+
+	if errSchema.String() != "" {
+		return EphemeralDB{}, fmt.Errorf("Error in populating schema in database: %s", errSchema.String())
+	}
+	if psqlErr != nil {
+		return EphemeralDB{}, fmt.Errorf("Error in populating schema in database: %s", psqlErr)
+	}
+	fmt.Printf("db schema executed in the %s database\n", newDBName)
+
+	database, err := connectToDatabase(true, newDBName, 6432)
+	if err != nil {
+		return EphemeralDB{}, err
+	}
+
+	dbq := &PostgreSQLDatabaseQueries{
+		dbConnection:   database,
+		allowTestUuids: false,
+		allowUnsafe:    true,
+	}
+
+	fmt.Printf("* WARNING: Unsafe PostgreSQLDB object was created. You should never see this outside of test suites, or personal development.\n")
+	res := EphemeralDB{db: dbq, dbContainerID: strings.TrimSpace(string(dockerContainerID)), dockerNetwork: tempNetworkName}
+
+	return res, nil
+}
+
+// NewEphemeralCleanTestFramework will clean a docker container,
+// and network. This is merely defined for testing purpose
+func NewEphemeralCleanTestFramework(dockerContainerID string, tempNetworkName string) error {
```

Comment:
I think this function signature can change to...
```
func (ephemeralDB *EphemeralDB) Dispose() {
}
```
... and the parameters that are currently passed, can be retrieved from `EphemeralDB` struct.

Reply:
Oh yes, silly me, this was the reason I changed the return signature, I'll update it :)